### PR TITLE
Fix bug where PersonDetailsTab still shows person after deletion

### LINKED ISSUE #142: Delete person not refreshing details panel
- Upcoming events still has deleted person's birthday on it 
- If details panel is showing full details of person, it does not switch back to the default panel if the person shown is deleted

### PR DESCRIPTION
**Pull request by chesterhow**
Fixes #142

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -41,6 +42,13 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deletePerson(personToDelete);
+
+        // TODO: check current tab and only switch if user is on PersonDetailsTab
+        if (model.getDetailedPerson().contains(personToDelete)) {
```
Comment (ivantjh) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:47]: Will the `model.getDetailedPerson()` be emptied out when switching to other panels?

Comment (ivantjh) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:47]: i.e. will the `list` be empty if the app is not showing the detailedPerson view?

Comment (chesterhow) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:47]: Nope. All this does is check if the detailedPerson is the personToDelete and if so, switches the tab to the upcomingEventsTab.

The limitation of such an implementation is that if they are not on the detailedPersonTab, (i.e. streaks tab) the details panel will still switch to upcomingEventsTab.

Comment (ivantjh) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:47]: Ok got it

**Review by ivantjh — APPROVED** [review]
lgtm!

**Pull request merged**